src/js/UI.js

```diff
@@ -107,7 +132,18 @@ export default class UI {
         const code = this.editor.getValue()
         //TODO: There is no API to cancel promies.
         //      Needs to think on strategy to do it.
-        compile({code}).then((response) => this.output(response.status, response.program_output))
-                       .catch((error) => this.output('255', error))
+
+        compile({code})
+            .then((response) => {
+                console.log(response)
+                if (response.program_output)
```

Comment:
I believe it would be better to move `if ... else ...` into a separate function.

and it would be better to have `this.toogle_run()` in one place instead of two